stdlib/public/Concurrency/AsyncCompactMapSequence.swift

```diff
@@ -0,0 +1,80 @@
+////===----------------------------------------------------------------------===//
+////
+//// This source file is part of the Swift.org open source project
+////
+//// Copyright (c) 2021 Apple Inc. and the Swift project authors
+//// Licensed under Apache License v2.0 with Runtime Library Exception
+////
+//// See https://swift.org/LICENSE.txt for license information
+//// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+////
+////===----------------------------------------------------------------------===//
+
+import Swift
+
+extension AsyncSequence {
+  @inlinable
+  public __consuming func compactMap<ElementOfResult>(
+    _ transform: @escaping (Element) async -> ElementOfResult?
+  ) -> AsyncCompactMapSequence<Self, ElementOfResult> {
+    return AsyncCompactMapSequence(self, transform: transform)
+  }
+}
+
+@frozen
+public struct AsyncCompactMapSequence<Upstream: AsyncSequence, ElementOfResult> {
+  public let upstream: Upstream
+  public let transform: (Upstream.Element) async -> ElementOfResult?
```

Comment:
the one thing that the upstream does have advantage of being public (since it is let) it can provide diagnostic information to debugging and such. will making it internal prevent us from building tooling around this?

Reply:
Not at all -- `@usableFromInline internal` members are still accessible via lldb or reflection. 

`@frozen` means the layout will be etched in stone anyway. (Is that a good idea for these?)

Reply:
made them internal to reduce API surface anyhow, and removed the `@frozen`